lr_face/data.py

```diff
@@ -743,6 +772,49 @@ def make_pairs(data: Union[Dataset, List[FaceImage]],
     return res
 
 
+def make_pairs_from_two_lists(
+        data_first: List[FaceImage],
+        data_second: List[FaceImage],
+        n: Optional[int] = None) -> List[FacePair]:
+    """
+    Takes two list of `FaceImage` instances and pairs them up, each pair
+    having one image from the first, one from the seconds
+
+    Returns:
+        A list of `FacePair` instances.
+    """
+    images_first_by_identity = defaultdict(list)
+    for x in data_first:
+        images_first_by_identity[x.identity].append(x)
+
+    images_second_by_identity = defaultdict(list)
+    for x in data_second:
+        images_second_by_identity[x.identity].append(x)
+
+    res = []
+    for identity in set(images_first_by_identity.keys()).intersection(
+            set(images_second_by_identity.keys())):
+        for image_a in images_first_by_identity[identity]:
+            for image_b in images_second_by_identity[identity]:
+                res.append(FacePair(image_a, image_b))
+
+    # Loop over and unpack all positive pairs, then create matching
+    # negative pairs that contain at least one of the images from each
+    # positive pair and a randomly chosen other image with a different
+    # identity.
+    for first, second in res.copy():  # Copy, because we modify `res`.
+        identity = first.identity
+        options = set(images_second_by_identity.keys()) - {identity}
+        if len(options) > 0:
+            negative_id = random.choice(tuple(options))
```

Comment:
it would be possible to pick the same negative image multiple times (maybe especially when using many categories). Do you think that is a problem?

Reply:
Niet heel erg. als er veel opties zijn is de kans klein, als er weinig opties zijn hebben we weinig keus. kan mooier, lage prio denk ik.